app/Http/Controllers/PostController.php

```diff
@@ -85,9 +87,16 @@ public function show($id)
      * @param  \App\Models\Post  $post
      * @return \Illuminate\Http\Response
      */
-    public function edit(Post $post)
+    public function edit($id)
     {
-        //
+
+        $post = Post::where('id', $id)->first();
```

Comment:
use association query to check a post belongs to current user or not

$currentUser->posts->where(...)